oop1.rb

```diff
@@ -0,0 +1,87 @@
+class Station
+	attr_reader :trains, :nameStation
+
+	def initialize(nameStation)
+		@nameStation = nameStation
+		@trains = []
+	end
+
+	def add_train(train)
+		@trains << train
+	end
+
+	def send_train(train)
+		@trains.delete_if{|elem| elem == train}
+	end
+
+end
+
+class Route
+	attr_reader :allStations
+	def initialize(firstStation, lastStation)
+		@allStations = []
+		@allStations[0] = firstStation
+		@allStations[1] = lastStation
+		@array_index = 0	
+	end	
+
+	def add_station(station)
+		@allStations << station
+		@array_index += 1
+		time_station = @allStations[@array_index]
+		@allStations[@array_index] = @allStations[@array_index + 1]
+		@allStations[@array_index + 1] = time_station
+	end
+
+	def remove_station(station)
+		@allStations.delete_if{|elem| elem == station}
+	end
+
+	def all_stations
+		@allStations.each {|station| puts station.nameStation}
+	end
+end
+
+class Train
+	attr_accessor :speed
+	attr_reader :carriages, :station, :last_station, :next_station, :type
+
+	def initialize(number, type, carriages)
+		@number = number
+		@type = type #passenger or freight
+		@carriages = carriages
+		@station = ""
+		@speed = 0
+	end
+
+	def stop 
+		@speed = 0
+	end
+
+	def change_carriages(add)
+		if @speed == 0
+			add == "+" ? @carriages+=1 : @carriages-=1
+		end
+	end
+
+	def add_route(route)
+		@route = route.allStations
```

Comment:
```suggestion
		@route = route
```
лучше так, чтобы не было путанницы
в переменной с именем "маршрут" лежит объект класс Маршрут - все предельно прозрачно